### PR TITLE
Java: Add models for android.net.Uri[.Builder]

### PR DESCRIPTION
**Pull request by smowton**

### CODE REVIEW

File: java/ql/test/library-tests/frameworks/android/uri/test.ql [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.ExternalFlow
+import semmle.code.java.dataflow.TaintTracking
+import TestUtilities.InlineExpectationsTest
```
Comment (atorralba) [inline java/ql/test/library-tests/frameworks/android/uri/test.ql:5]: `java/ql/test/TestUtilities/InlineFlowTest.qll` can be used imported of the boilerplate code.

**Review by atorralba — COMMENTED** [review]
A couple of inline comments, otherwise LGTM.

File: java/ql/lib/semmle/code/java/frameworks/android/Android.qll [hunk 2]
```diff
@@ -79,3 +80,71 @@ class AndroidContentResolver extends AndroidComponent {
...
   }
 }
+
+private class UriModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "android.net;Uri;true;buildUpon;;;Argument[-1];ReturnValue;taint",
+        "android.net;Uri;false;decode;;;Argument[0];ReturnValue;taint",
+        "android.net;Uri;false;encode;;;Argument[0];ReturnValue;taint",
```
[the remark this answers is not in the document]
Comment (smowton) [reply java/ql/lib/semmle/code/java/frameworks/android/Android.qll:90]: Yes

File: java/ql/test/library-tests/frameworks/android/uri/test.ql (hunk 1 above)
Comment (smowton) [reply java/ql/test/library-tests/frameworks/android/uri/test.ql:5]: Done

**Review by atorralba — APPROVED** [review]
Approving because I'm quite confident this is OK, but feel free to wait for further reviews if you think it's needed.

**Pull request merged**